tests/lib/modem_trace/CMakeLists.txt

```
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(modem_trace_test)
+
+# create mock
+cmock_handle(${ZEPHYR_BASE}/../modules/hal/nordic/nrfx/drivers/include/nrfx_uarte.h )
+cmock_handle(${ZEPHYR_BASE}/../modules/debug/segger/SEGGER/SEGGER_RTT.h )
+
+# generate runner for the test
+test_runner_generate(src/modem_trace_test.c)
+
+target_include_directories(app PRIVATE src)
+
+# add test file
+target_sources(app PRIVATE src/modem_trace_test.c)
+
+# add unit under test
+target_sources(app PRIVATE ../../../lib/nrf_modem_lib/modem_trace.c)
+
+# include paths
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../modules/hal/nordic/nrfx)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../modules/hal/nordic/nrfx/mdk)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../modules/hal/cmsis/CMSIS/Core/Include/)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../modules/debug/segger/Config/)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/modules/hal_nordic/nrfx)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../nrf/include/modem/)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../nrfxlib/nrf_modem/include/ )
+
+
+# Define the device. Otherwise an "unknown device" warning will be triggered when compiling
+# nrfx_config.h
+target_compile_options(app PRIVATE -DNRF9160_XXAA)
```

minor nit, `target_compile_definitions()` exists specifically for defines.
```suggestion
target_compile_definitions(app PRIVATE NRF9160_XXAA)
```

same goes for lines below.